packages/teleport-generator-core/src/validator/index.ts

```diff
@@ -38,6 +42,66 @@ export default class Validator {
 
     return { valid: true, errorMsg: '' }
   }
+
+  public validateComponentContent(input: any): ValidationResult {
+    const verifyProps = checkContent(input, 'prop')
+    const verifyState = checkContent(input, 'state')
+
+    checkForDuplicates(input)
+    const errors = verifyProps.errors.concat(verifyState.errors)
+
+    if (errors.length > 0) {
+      return {
+        valid: false,
+        errorMsg: `\nUIDL Content Validation Error. Please check the following: \n${errors}`,
+      }
+    }
+
+    return { valid: true, errorMsg: '' }
+  }
+}
+
+const checkForDuplicates = (input: any) => {
+  const props = Object.keys(input.propDefinitions || [])
```

Comment:
the default value for `input.propDefinitions` should be an empty object, not empty array